ConsoleApplication2/ConsoleApplication2/ConsoleApplication2.cpp

```diff
@@ -0,0 +1,259 @@
+// ConsoleAplenplication2.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
+
+#include <iostream>
+#include <exception>
+#include <algorithm>
+
+#include <functional>
+#include <array>
+#include <string_view>
+
+
+#define ALLOCATE_BYTES    10
+
+typedef unsigned int index_t;
+
+class SString {
+public:
+    SString() {
+        this->str = nullptr;
+        this->len = 0;
+        this->cap = 0;
+    }
+
+    SString(const char* in) : SString() {
+        while (in[this->len] != '\0') {
+            this->len++;
+        }
+        if (this->len > 0) {
+            reallocStorage(this->len + 1);
+
+            copyString(this->str, in, this->len + 1);
+        }
+    }
+
+    SString(const SString& in) : SString() {
+        this->len = in.len;
+        if (this->len > 0) {
+            reallocStorage(this->len + 1);
+
+            copyString(this->str, in.str, this->len + 1);
+        }
+    }
+
+    SString(SString&& in) : SString() {
+        this->len = in.len;
+        if (this->len > 0) {
+            reallocStorage(this->len + 1);
+
+            copyString(this->str, in.str, this->len + 1);
+        }
+
+        in.str = nullptr;
+    }
+
+    ~SString() {
+        std::cout << "memory free" << std::endl;
+        delete[] this->str;
+    }
+
+    // return curent symbol from string
+    char& operator[] (index_t ind) {
+        return this->str[ind];
+    }
+
+    // return current symbol from string with access of range
+    char& at(index_t ind);
+
+    void append(const char* in);

```

Comment:
Let's have append methods returning SString& (as copy c-tor does) so we can write sth like
str.append("Hello").append(" world").append(".");